api/app.py

```diff
@@ -17,69 +18,169 @@
 
 @app.route('/')
 def hello_word():
-    return 'Welcome to my snack app!'
+    return 'Welcome to my snack app!', 200
 
 
-@app.route('/api/recipes', methods=['GET'])
+@app.route('/api/recipes/', methods=['GET'])
 def get_recipes():
     recipes = []
 
     for recipe in db.session.query(Recipe).all():
         recipes.append(recipe.as_dict())
 
-    return jsonify(recipes)
+    return jsonify(recipes), 200
 
 
 @app.route('/api/recipes/<int:recipe_id>', methods=['GET'])
 def get_recipe(recipe_id):
-    #for recipe in recipes:
+    # for recipe in recipes:
     #    if recipe['id'] == recipe_id:
     #        return jsonify(recipe)
-    #return jsonify({'error': "404 Not found"}), 404
+    # return jsonify({'error': "404 Not found"}), 404
 
     for recipe in db.session.query(Recipe).all():
         recipe = recipe.as_dict()
         if recipe['id'] == recipe_id:
-            return jsonify(recipe)
+            return jsonify(recipe), 200
     return jsonify({'error': "404 Not found"}), 404
 
+
 @app.route('/api/recipes', methods=['POST'])
 def create_recipe():
-    new_recipe = {
-        'id': len(recipes) + 1,
+    # Get data from the request
+    recipe_data = {
         'name': request.json.get('name'),
         'duration': request.json.get('duration'),
         'pictures': request.json.get('pictures'),
         'instructions': request.json.get('instructions'),
         'ingredients': request.json.get('ingredients'),
         'categories': request.json.get('categories'),
     }
-    recipes.append(new_recipe)
-    return jsonify(new_recipe), 201
+
+    missing_fields = []
+    for key, item in recipe_data.items():
+        if not item:
+            missing_fields.append(key)
+
+    if len(missing_fields) != 0:
+        return jsonify({'error': 'Missing fields',
+                        'Missing': missing_fields}), 400
+
+    category_obj = []
+    for cat in recipe_data['categories']:
+        category = Category.query.filter_by(id=cat['name']).first()
+        if not category:
```

Comment:
We don't need to create a `category` that doesn't exist here; we'll use the `create_category()` endpoint. Here you can return an error message if the category does not exist, for example:
```python
return jsonify({'error': f'category {cat["name"]} not found'}), 400
``` 
Same for `edit_recipe()`.